lib/srv/db/cloud/watchers/rds.go

```diff
@@ -196,5 +202,74 @@ func auroraFilters() []*rds.Filter {
 	}}
 }
 
+// isDBClusterSupported checks whether the aurora cluster is supported and logs
+// related info if not.
+func isDBClusterSupported(cluster *rds.DBCluster, log logrus.FieldLogger) bool {
+	switch aws.StringValue(cluster.EngineMode) {
+	// Aurora Serverless (v1 and v2) does not support IAM authentication
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-2.limitations.html
+	case RDSEngineModeServerless:
+		log.Infof("Aurora cluster %q is %s which doesn't support IAM authentication. Skipping.",
+			aws.StringValue(cluster.DBClusterIdentifier),
+			aws.StringValue(cluster.EngineMode),
+		)
+		return false
+
+	// Aurora MySQL 1.22.2, 1.20.1, 1.19.6, and 5.6.10a only: Parallel query doesn't support AWS Identity and Access Management (IAM) database authentication.
+	// https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-mysql-parallel-query.html#aurora-mysql-parallel-query-limitations
+	case RDSEngineModeParallelQuery:
+		if utils.SliceContainsStr([]string{"1.22.2", "1.20.1", "1.19.6", "5.6.10a"}, auroraMySQLVersion(cluster)) {
+			log.Infof("Aurora cluster %q (engine mode %v, engine version %v) doesn't support IAM authentication. Skipping.",
+				aws.StringValue(cluster.DBClusterIdentifier),
+				aws.StringValue(cluster.EngineMode),
+				aws.StringValue(cluster.EngineVersion),
+			)
+			return false
+		}
+
+	// other modes are compatible
+	default:
+	}
+
+	return true
+}
+
+// auroraMySQLVersion extracts aurora mysql version from engine version
+func auroraMySQLVersion(cluster *rds.DBCluster) string {
+	// version guide: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Updates.Versions.html
+	// a list of all the available versions: https://docs.aws.amazon.com/cli/latest/reference/rds/describe-db-engine-versions.html
+	//
+	// some examples of possible inputs:
+	// 5.6.10a
+	// 5.7.12
+	// 5.6.mysql_aurora.1.22.0
+	// 5.6.mysql_aurora.1.22.1
+	// 5.6.mysql_aurora.1.22.1.3
+	//
+	// general format is: <mysql-major-version>.mysql_aurora.<aurora-mysql-version>
+	// 5.6.10a and 5.7.12 are "legacy" versions and they are returned as it is
+	parts := strings.Split(aws.StringValue(cluster.EngineVersion), ".")
+	for i, part := range parts {
+		if part == "mysql_aurora" {
+			return strings.Join(parts[i+1:], ".")
+		}
+	}
+	return aws.StringValue(cluster.EngineVersion)
```

Comment:
Consider using `strings.Index` instead of `Split/Join`.

```suggestion
	const tag = "mysql_aurora"
	version := aws.StringValue(cluster.EngineVersion)
	if i := strings.Index(version, tag); i != -1 {
		if startIdx := i + len(tag) + 1; startIdx < len(version) {
			return version[startIdx:]
		}
	}
	return version
```

Reply:
Personally I don't like index calculation unless it's necessary. Is there any particular case you have in mind that it's missing in original implementation?

I do think Split makes logic easier to understand. That said, I probably can do split on `.mysql_aurora.`, and if `len(parts) == 2`, then return `parts[1]`. I can probably look into this.

Performance wise Split/Join definitely uses more memory and cpu but it's probably in `ns` difference and this is not on critical path 😄

Reply:
Splitting on `.mysql_aurora.` is a good idea. 

As you said, performance isn't really a concern here, but FWIW, there are no allocations for the `Index` version (vs ~8 for split) and `Index` is ~16x faster on my machine using the examples in your comments.

I commented on this because it was more about understanding what it was doing. It took me a few glances because I didn't expect to see a split/join given the description. Had I seen `Index`, I feel like I would've understood right away since `Index` is designed to search for substrings. But accuracy is more important and both versions do the job.

Reply:
Looks like Go 1.18 will add a new [`Cut` function](https://github.com/golang/go/issues/46336) that will make common tasks like this even better.  🙂

```
_, v, ok = strings.Cut(version, ".mysql_aurora.")
if ok {
  return v
}
```